tests/misc.rs

```diff
@@ -17,9 +17,11 @@
 // extern crate elf;
 // use std::path::PathBuf;
 
-extern crate rbpf;
+// These tests should only be run on std as they depend on the assembler.
+#![cfg(feature = "std")]
```

Comment:
Do they? It seems a lot of them don't need the assembler. Plus most of them seem already gated behind individual `std` directives?

Reply:
Good point, I will enable a more fine-grained control so that everything that can possibly be run runs when testing without std